### PR TITLE
wider default  fsharp_max_value_binding_width, also take width of pattern of binding into account

### PR DESCRIPTION
**Pull request by dsyme**
Adjusts the default `fsharp_max_value_binding_width` up to 80, see https://github.com/fsprojects/fantomas/issues/1933

I adjusted things computation to take into account the width of the pattern of the let (`pattern = expr`), see comment below and adjustment to docs.  This feels right to me.

@nojaf It feels right to make this change

@Smaug123 Could you find what G-Research default setting is?  May make sense to simply go to that?

### CODE REVIEW

**Comment by dsyme** [comment]
@nojaf @Smaug123 I adjusted this to include the pattern in "let pattern = expr" in determining "fsharp_max_value_binding_width".  This feels right, e.g.

```fsharp
let aaaa, bbbb, cccc = expr
```
will split to 
```fsharp
let aaaa, bbbb, cccc =
    expr
```
but
```fsharp
let a = expr
```
will not (for suitable length of expr etc.)

**Comment by dsyme** [comment]
@nojaf This is ready for review.

**Comment by nojaf** [comment]
Hey Don, sorry for this but would you mind targeting the [4.6 branch](https://github.com/fsprojects/fantomas/tree/4.6) for this PR? 
There was no way for you to know this, my bad.

The idea is that bug fixes are made against master and revisions are published from there.  
For new features or grander stylistic changes, the feature branch should be used.
I'll add this to the contribution guidelines next week. 
Sorry again, I hope you can cherry-pick the commits easily.

**Comment by dsyme** [comment]
> Hey Don, sorry for this but would you mind targeting the 4.6 branch for this PR?

No probs, done. Do you manually merge master --> 4.6?

FWIW this is the opposite to dotnet/fsharp.  There we do all feature work in master, and set up codeflows

    master    // for feature work and bug fixes 
    release/4.6   // this is the "next" release. 
    release/4.5   // this is a previous release  
    release/4.4   // this is a previous release  

Code flow prior to releasing 4.6:

    master --> release/4.6   // this is the "next" release. feature work and bug fixes flow to next release unless targeted at existing release
    master <-- release/4.5   // bug fixes can be made in previous releases and will flow to master and future release

Code flow after releasing 4.6:

    master --> release/4.7   // this is the "next" release. 
    master <-- release/4.6   // bug fixes can be made in previous release and will flow to master and future release

The code flow is automated and creates integration PRs. I really wish github did this by default for all projects, and you could put a ".github/codeflow/config.yml" for active flows.

**Comment by Smaug123** [comment]
Nothing is set in stone here as far as G-Research is concerned; due to coordination difficulties and the lack of time to resolve them, the things we've got formatted internally are sufficiently low-traffic that we would happily reformat them if it meant standardising on a "better" model of the world. I think this kind of line-length optimisation is probably one of the things we have the least strong opinions on.

**Comment by nojaf** [comment]
> No probs, done. Do you manually merge master --> 4.6?

Thanks, yes I admit, I do this manually right now. 4.6 is the first version where we actually try and maintain two branches.
In the past, we did everything on master and from the first new unstable feature or FCS upgrade we labelled it alpha. Fixes went into the alpha but people were hesitant to try out these versions.

Now, we have released five revisions (4.5.X) and we do see a spike in downloads on NuGet.
People are more comfortable with this model.
Which, is a long way of telling that we are still figuring this out.
Thanks for the ` ".github/codeflow/config.yml"`, I'll definitely check this out sometime.

File: fantomas.sln [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
+# Visual Studio Version 17
+VisualStudioVersion = 17.0.31825.309
 MinimumVisualStudioVersion = 15.0.26124.0
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = ".paket", ".paket", "{B570C54C-2FEA-4A94-A8C5-FD4157A94DE7}"
 	ProjectSection(SolutionItems) = preProject
 		paket.dependencies = paket.dependencies
 	EndProjectSection
 EndProject
-Project("{F2A71F9B-5D33-465A-A702-920D77279786}") = "Fantomas", "src\Fantomas\Fantomas.fsproj", "{7EA16279-A5F1-4781-A343-4375A04BCE6F}"
+Project("{6EC3EE1D-3C4E-46DD-8F32-0CC8E7565705}") = "Fantomas", "src\Fantomas\Fantomas.fsproj", "{7EA16279-A5F1-4781-A343-4375A04BCE6F}"
```
Comment (nojaf) [inline fantomas.sln:11]: What does this mean? Or did this happen automatically by VS?

File: src/Fantomas/CodePrinter.fs [hunk 2]
```diff
@@ -4953,6 +4953,13 @@ and genMemberDefn astContext node =
                 | None -> sprintf "abstract %s" s
             |> fun s -> !- s ctx
 
+        let hasGenerics =
```
Comment (nojaf) [inline src/Fantomas/CodePrinter.fs:4956]: Could you try and avoid having a PR that does two unrelated things?
I'm ok with this change, but it seems unrelated to changing the default settings.

Comment (nojaf) [reply src/Fantomas/CodePrinter.fs:4956]: It is the same fix as https://github.com/fsprojects/fantomas/pull/1935, right?

File: docs/Documentation.md [hunk 3]
```diff
@@ -729,7 +730,7 @@ type MyType() =
 
 ### fsharp_max_function_binding_width
 
-Control the maximum width for which let and member function bindings should be in one line.
+Control the maximum width for which function and member bindings should be in one line.
```
Comment (nojaf) [inline docs/Documentation.md:733]: Member bindings will look at two settings currently.
[Example](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAYrLAEYCGATgBQCUYAvADrkMB2EEA9B2AGYDOACwrYA%2BgFtSADxEA3UgBsArohHEAli3gaA5iIDua%2BOgF0AjE1bswYxGOKJyYY2r4A6ABr0wG9BbbsuXkFhcSkRHkUWaHQ1WBZVDS0WXQMjEwAWc2Z-CBs7BycBF1cATTAaLx8-SAAacCg%2BRHQY5L5UMABteohQKwhIABIsXHbIH1ru9kG5JWQ0Lr72AAY6xf6ASU1EFnQAZTUAL2RVtfTJiABdSYBfE57zqCGcOahxu6mBmeV2hbXTd6skAAslIADIaRCg7baYwTNZgUwAJiWDyuVluk16fUGwxekGIRHkcOxnwU33mDwgiIBU12tjUcHkcQAgugAKKaADyPHBLGOlN4CkaqJu7yxgKeIzQ%2BMJxIlXxev0WAGYaf1dthSNBEAAhRA8WDkRAABQopBs6Accr66HIyhF6LFDxxz1GBNgRJp0zJioF6TVUA1Wt1%2BsNkNgegc0FIjU2Mlg0ZicWtVlt9sWaPYGKs4o%2BuLdsq9pNmPwFAFYA5Ag9q9QajQBVbC4cjR2MseOJ2IsFPsHhCxAOrNOxYuqVQd2e53F8mdAUANkr1ZDdcQAGF5DG%2BKu4nw09FDT2IH35MKM6LMVP89KJz3vSWKfCAOyLzU10NGoG2ezkQ%2BCk8Ds9HQvEdJTxG8iwVUt4QADhfYNazDbcmW7ANj1PPpMwgbN2FzfpQILD1b2nX14QATjg7VmR4S1yG3MQJF-NMAIw88c0vV1r0LKdIIfP4VgFKtX2XMM6TEBkPWTVD%2B0HLDhxJK9xy4kCeNneFTH%2BASlyomjRPE5DGLtZirEwsBsPuECFJlQiIJ9KC-mpATNiQHZORYAAVchMAAdTUWEpP-GTTLkiVLPA7jbN4xZTFVTShOZchYEieAABFEHkNQxJogz0xYoC2IsjjXh2IiVKVPpTH9ASQUkdYeDcgRtjZf9dgEQ10B8tJf3SFFAKHYD5MKsZips%2B9VL%2BCsqqkTYeDUSROWbUh0ENNlJGwI0%2BD4LtfzLHrcr6-KBrHIb0BKiKxqihdJskAAlRA4HIeAOr8v1duM1icPYo63nC0ayqsUxnyu277vgAA5RR8nIbl1ktMQ2gDUxArMsBcMeSyxEUeQYnSvlCHICRmgcNzXRGmc-vYUxYIE4HDXgIFMexiE8YJrLK2gIRyC1Gj9EMWEkeCvNBu%2B5SzvJiBTHIq74s5zBOXIcFdyegQute9gTOR1HRzxYWSVKgVkUrarpdIWX5ZcdBwch6HYfhgVEd62T%2BpCwaMaxtQccQZnFpo4mRlJki1kRDT4UgY3TYV9B6bdj2vcJn82Y5rmHB5zr%2BadwWvuGn6yf1hyQ%2BqgA1H0dUSHQld-e29sdg7nczk7-bsxZERi-OpHwSJoi7EvNDL3nlYDTI05rjPtazkXfv1yrW8kZLYHQABxJoVrWxANq7cuAx2oePoKo6wvHnP4URCb84Z92IR1JloAAax1Tmb6aPhXN2c01w9CGUIFNCjLV97zMOsCSldai31pdEOoNEB6A9nqdAkZti%2B0QKlGaLBfJdmZJoT8kNbbwm-tvf%2BtdAHWWzgHJugMQ4AGlECIGwLVeq2xNgvxsLyfkODpIOyCunPClkdbyhAUfKm08mpiUEBvF6eCUafUIZOA%2BJC%2BiIkliHfYyR5CIHitoCG2x2qIGIECBALC1i4PYRrSRBFpHAInvCZU-EQ7MnStoFg7cohJhYPsexi1FBGjcrAJymjFpbX8uhN6eUd4ANMadCxaxlTBzWKHLGDgWB%2BJkOGZImDvxII0Kgncv5DFVw4cPLhg197mMPpEvOMSo4xGYaCc0xB4CkHXLATayQIFQIhNkthuTjG7ykeEkpKoW4xOSi4UgxAVFCJcAIXYmAdhSHaQFIxAsClHW2PAbkzDdhYBUb02RVhlRTxiRyVZPI2ls3IPIHgkBxGa3wpxIhMjG59GVCfGJlDqE%2BJ2JsO%2BpAoj9y-h0oJ%2B0QkELCQ3SKjywExMvt86%2BzC%2BDxUSpoCBu5EB0zPh7Ze61NpZIDExK5JjblmN4RElUZDIUUAQkaIZfBoDkAyhoRaKL6woLiKlaAG5ObOLmYE3%2BwT8EjxBcQh5uyBExI2bS6IuikBcp-pcG49QLgnDGHwfAm12i4OuEAA).

**Comment by nojaf** [comment]
Closing in favour of #1947. Thanks again Don.
